components/movementsensor/collectors.go

```diff
@@ -19,31 +49,162 @@ func assertMovementSensor(resource interface{}) (MovementSensor, error) {
 	return ms, nil
 }
 
-type lowLevelCollector func(ctx context.Context, ms MovementSensor, extra map[string]interface{}) (interface{}, error)
+func newPositionCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
```

Comment:
Just so I understand, you did not add new collectors for the movement sensor, but rather moved the collectors from `movementsensor.go` to here (`collectors.go`)?

And then ofc made the change to return the proto response format.